xmppserver/src/main/java/org/jivesoftware/openfire/streammanagement/StreamManager.java

```diff
@@ -346,6 +346,12 @@ private void startResume(String namespace, String previd, long h) {
             return;
         }
         Log.debug("Found existing session for '{}', checking status", fullJid);
+
+        // OF-2811: Cannot resume a session that's already closed. That session is likely busy firing its 'closeListeners'.
+        if (route.isClosed()) {
+            Log.debug("Not allowing a client of '{}' to resume a session, as the preexisting session is already in process of being closed.", fullJid);
```

Comment:
Did you want to send an error and return here?

Reply:
:facepalm: